src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java

```diff
@@ -94,15 +105,20 @@ public BlobStoreContext getContext() throws IOException {
         // get user credentials from env vars, profiles,...
         AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
         // Assume we are using session credentials
-        AWSSessionCredentials awsCredentials = (AWSSessionCredentials) builder.getCredentials().getCredentials();
+        AWSCredentials awsCredentials = builder.getCredentials().getCredentials();
+
         if (awsCredentials == null) {
             throw new IOException("Unable to get credentials from environment");
         }
 
+        if(!(awsCredentials instanceof AWSSessionCredentials)){
+            throw new IOException("No valid session credentials");
```

Comment:
Is this related somehow?

Reply:
it is another error found in the process if your session does not have token the credential type is different and you will see a cast exception

Reply:
I mean, is this an issue that arises when adding the ability to select a region, or just something you noticed along the way?

Reply:
I will open other jira

Reply:
Not necessary I guess, just wanted to be clear whether it _could_ have been in a separate PR or whether it was really intrinsically part of this one.

Reply:
It could be in another PR, it is not needed for this issue.